examples/cloudflare-workers/http-api/src/worker.ts

```diff
@@ -13,11 +13,8 @@ class MomentoFetcher {
 	private readonly baseurl: string;
 	constructor(key: string, endpoint: string) {
 		this.apiKey = key;
-		if (!endpoint.startsWith('https://')) {
-			this.baseurl = `https://${endpoint}/cache`;
-		} else {
-			this.baseurl = `${endpoint}/cache`;
-		}
+		const correctedEndpoint = endpoint.replace(/^(http:\/\/)?/, "https://");
```

Comment:
thanks for the submission!

I don't think this patch correctly handles the case where the URL already begins with `https://`, and I'm also not 100% sure whether we want to forcefully change the URL to `https://` if the caller explicitly gave us `http://`?

```
 npx ts-node
> "foo.bar.com".replace(/^(http:\/\/)?/, "https://");
'https://foo.bar.com'
> "https://foo.bar.com".replace(/^(http:\/\/)?/, "https://");
'https://https://foo.bar.com'
>
```

I think maybe another idea would be to do:

```typescript
if (!(endpoint.startsWith('https://') || endpoint.startsWith('http://')) {
...
```

That would leave either of the two protocols intact if a user passed them, while still adding `https://` prefix if they did not. WDYT?

Reply:
I agree that parsing the URL is probably a more correct way to do it. In another environment, I would definitely write the code the way you have here.

I suspect the full URL parsing is quite a bit more expensive than just a `.startsWith`, though, and in a CloudFlare environment I wonder if the perf difference could matter. That's probably why we did it this way initially.

The latency of the HTTP request itself probably dwarfs the URL parsing though, so what I said above is probably a micro-optimization :)

Momento doesn't currently support plaintext endpoints but it's theoretically possible we could someday. So overall my thoughts are:

* probably best not to change the protocol that the user expressed if they expressed one
* parsing the URL is definitely the most correct way to handle it
* perf implications of parsing the URL vs startsWith are probably not really a big deal.

So we can change it either to what you suggested above, or what I suggested in my prev comment, I would be fine with either!

Reply:
Excellent observations. Let's go with your suggestion of `if ( !( endpoint.startsWith('https://') || endpoint.startsWith('http://') ) ) {`

- I think it's clearer what's being tested in the conditional
- Rebuilding the original endpoint from the parsed URL components is quite verbose
- The opportunity to throw an error for an improperly formatted URL that was probably copy/pasted from an admin panel seems like an opportunity that will be seldom taken.

Do note that your inline code suggestion was lacking a third closing parenthesis, which is fixed in my inline code above.

Thanks for your time!

Reply:
sure thing, thanks for pointing out the issue!